sympy/matrices/decompositions.py

```diff
@@ -1363,7 +1363,7 @@ def dot(u, v):
             Q[:, j] -= Q[:, i] * R[i, j]
 
         Q[:, j] = dps(Q[:, j])
-        if Q[:, j].is_zero_matrix is False:
+        if Q[:, j].is_zero_matrix is not True:
```

Comment:
Ok, having experimented with the below, I agree with this change.

But perhaps it could use a comment:
```
  # if we are sure the column is zero, we do not need to include it
  if Q[:, j].is_zero_matrix in (False, None):
```
Or how about this without a comment?
```
  if Q[:, j].is_zero_matrix:
       continue
  ranked.append(j)
  R[j, j] = M.one
```